src/ott/geometry/geodesic.py

```diff
@@ -0,0 +1,308 @@
+# Copyright OTT-JAX
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any, Dict, List, Optional, Sequence, Tuple
+
+import jax
+import jax.numpy as jnp
+import numpy as np
+from jax.experimental.sparse.linalg import lobpcg_standard
+from scipy.special import ive
+
+from ott.geometry import geometry
+from ott.math import utils as mu
+
+__all__ = ["Geodesic"]
+
+
+@jax.tree_util.register_pytree_node_class
+class Geodesic(geometry.Geometry):
```

Comment:
Please add this class to the `docs/geometry.rst`.

Reply:
Added.